raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -609,7 +631,8 @@ private int binaryExponentialElectionBackoffMs(int retries) {
             throw new IllegalArgumentException("Retries " + retries + " should be larger than zero");
         }
 
-        return Math.min(RETRY_BACKOFF_BASE_MS * random.nextInt(2 << (retries - 1)), electionBackoffMaxMs);
+        // upper limit exponential co-efficients at 16
+        return Math.min(RETRY_BACKOFF_BASE_MS * random.nextInt(2 << Math.min(16, retries - 1)), electionBackoffMaxMs);
```

Comment:
We may go beyond in.max and causing negative timeout, fixing here to upper bound the exponential co-efficient.

Reply:
I hit the same bug and chose 20 arbitrarily in #377 🙂 .